#!/usr/bin/env python
PACKAGE = "travesim_adapters"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Unicast addresses must be in the block 127.0.0.0/8. See:
# https://www.iana.org/assignments/iana-ipv4-special-registry/iana-ipv4-special-registry.xhtml

# User Ports (1024 - 49151). See:
# https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml

gen.add("yellow_team_address", str_t, 0, "Yellow Team Address (127.0.0.0 - 127.255.255.255)",  "127.0.0.1")
gen.add("yellow_team_port", int_t, 0, "Yellow Team Port", 20012,  1024, 49151)

gen.add("blue_team_address", str_t, 0, "Blue Team Address (127.0.0.0 - 127.255.255.255)",  "127.0.0.1")
gen.add("blue_team_port", int_t, 0, "Blue Team Port", 20013,  1024, 49151)

gen.add("specific_source", bool_t, 0, "Specific Source for Teams",  False)
gen.add("reset", bool_t, 0, "Reset teams adapter",  False)

exit(gen.generate(PACKAGE, "teams_adapter", "Teams"))
